atd-vze/src/Components/GridTableSearch.js

```diff
@@ -0,0 +1,130 @@
+import React, { useState, useEffect } from "react";
+import { withApollo } from "react-apollo";
+import {
+  Col,
+  Button,
+  Form,
+  FormGroup,
+  Input,
+  InputGroup,
+  InputGroupAddon,
+  InputGroupButtonDropdown,
+  DropdownToggle,
+  DropdownItem,
+  DropdownMenu,
+  Alert,
+} from "reactstrap";
+
+// TODO add query operators to each field to better fit data types (_eq, etc.)?
+const GridTableSearch = ({ query, clearFilters, setSearchParameters }) => {
+  const [searchFieldValue, setSearchFieldValue] = useState("");
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+  const [fieldToSearch, setFieldToSearch] = useState("");
+  const [isFieldSelected, setIsFieldSelected] = useState(false);
+
+  const fieldsToSearch = query.searchableFields;
+
+  /**
+   * Handles the submission of our search form
+   * @param {object} e - the event object
+   */
+  const handleSearchSubmission = e => {
+    e.preventDefault();
+
+    setSearchParameters({
+      column: fieldToSearch,
+      value: searchFieldValue,
+    });
+  };
+
+  /**
+   * Clears the search results
+   */
+  const handleClearSearchResults = () => {
+    clearFilters();
+    setSearchFieldValue("");
+    setFieldToSearch("");
+    setIsFieldSelected(false);
+  };
+
+  /**
+   * Toggles the dropdown options
+   */
+  const toggleDropdown = () => {
+    setIsDropdownOpen(!isDropdownOpen);
+  };
+
+  /**
+   * Handles the selection of our search mode in the dropdown
+   * @param {object} e - the event object
+   */
+  const handleFieldSelect = e => {
+    setIsFieldSelected(true);
+    setFieldToSearch(e.target.value);
+  };
+
+  /**
+   * Returns a human-readable label for a specific column
+   * @param {string} fieldKey - the raw column name from the database
+   * @returns {string}
+   */
+  const getFieldName = fieldKey => {
+    return query.getLabel(fieldKey, "search");
+  };
+
+  return (
+    <Form className="form-horizontal" onSubmit={handleSearchSubmission}>
+      {!isFieldSelected && searchFieldValue && (
+        <Alert color="warning">Please provide a field to search.</Alert>
+      )}
+      <FormGroup row>
+        <Col md="6">
+          <InputGroup>
+            <Input
+              type="text"
+              id="input1-group2"
+              name="input1-group2"
+              placeholder={"Enter Search Here..."}
+              value={searchFieldValue}
+              onChange={e => setSearchFieldValue(e.target.value)}
```

Comment:
I think we need to do more in this `onChange` event so its worth creating a `const handleSearchValueChange = () => {}` arrow function above. Since our DB expects addresses to be capital case, is there a way we could do some text transformation on the client? How do we deal with different casing for different search fields in this abstraction? 

In general, inline functions for click events was anti-pattern in React, but I wonder if it matters as much in the new React world hooks and functional stateless-components:
https://cdb.reacttraining.com/react-inline-functions-and-performance-bdff784f5578